### PR TITLE
created dropdown component

### LINKED ISSUE #7: Dropdown component
- [ ] Some props: disabled, hover

- [ ] Variations: primary, secondary, text button, icon 

- [ ] Add to Charcuterie page

### PR DESCRIPTION
**Pull request by rzyxie**
dropdown menu with ability to input a custom value if a particular option is selected

NOTE: not sure how we want the input field to look like on the app since we haven't discussed the ability to enter custom amounts for the tip, but it makes sense to have so i've added the feature anyways 

**Props:**
options: array of items to list in the dropdown 
defaultOption: the default option to display when dropdown is initially rendered; default is the first item in array
size: small, medium, large; default is medium 

**If we want to show an input field for custom values (i.e. a custom tip):** 
showInput/setShowInput: boolean state for the input field
customValue: the option value that will trigger the input field to be rendered 

for example, when a user selects the "Other" option to enter a custom tip, an input field will show for a user to input a custom value

image

Figma design for context:
image

### CODE REVIEW

File: src/components/Dropdown.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+font-weight: ${(p) => p.theme.fontWeight};
+color: ${(p) => p.theme.colors.text};
+padding: 10px;
+border-radius: 5px;
+margin: 0.25em;
+padding: 0.25em 0.25em;
+`;
+
+const Dropdown = ({ options, defaultOption, size, showInput, setShowInput, customValue, ...props }) => {
+    const [selected, setSelected] = useState(defaultOption);
```
Comment (christylo) [inline src/components/Dropdown.js:17]: love to see the hook <3

File: src/components/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const handleSelect = (event, setSelected, setShowInput, customValue) => {
+    let selectedValue = event.target.value;
+    setSelected(selectedValue);
+    if (setShowInput) {
+        if (selectedValue === customValue) {
+            setShowInput(true);
+        } else {
+            setShowInput(false);
+        }
```
Comment (christylo) [inline src/components/Dropdown.js]: could use a ternary operator for cleaner code?
```suggestion
        let showInputVal = selectedValue === customValue ? true : false;
        setShowInput(showInputVal);
```

**Review by christylo — APPROVED** [review]
Looks good! About your note, maybe we can just keep it to dropdown tip value instead of custom for our mvp and add in the other option later on. Good to have it handy for later use though thanks!!

File: src/components/Dropdown.js (hunk 2 above)
Comment (rzyxie) [reply src/components/Dropdown.js]: yesyes very clean thank you!

File: src/components/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        </div>
+    )
+}
+
+const handleSelect = (event, setSelected, setShowInput, customValue) => {
+    let selectedValue = event.target.value;
+    setSelected(selectedValue);
+    if (setShowInput) {
+        let showInputVal = (selectedValue === customValue) ? true : false;
+        setShowInput(showInputVal);
```
Comment (kozr) [inline src/components/Dropdown.js]:
```suggestion
       setShowInput(selectedValue === customValue)
```

File: src/components/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+}
```
Comment (kozr) [inline src/components/Dropdown.js]: You can put this method inside the functional component, that way you don't have to pass in the hooks. It's only used there anyway.

e.g. cut and paste this and put it between line 17 and 18

File: src/components/Dropdown.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+border-radius: 5px;
+margin: 0.25em;
+padding: 0.25em 0.25em;
+`;
+
+const Dropdown = ({ options, defaultOption, size, showInput, setShowInput, customValue, ...props }) => {
+    const [selected, setSelected] = useState(defaultOption);
+    return (
+        <div>
+            <DropdownSelect size={size} value={selected} onChange={(e) => handleSelect(e, setSelected, setShowInput, customValue)}>
```
Comment (kozr) [inline src/components/Dropdown.js]: Update the handleSelect param in response to the handleSelect method changing places

**Review by kozr — CHANGES_REQUESTED** [review]
.

File: src/components/Dropdown.js (hunk 4 above)
Comment (rzyxie) [reply src/components/Dropdown.js]: ah i see what you mean - thanks for the suggestion, will do!

**Review by kozr — APPROVED** [review]
lgtm!!

**Pull request merged**